client/src/api/photos.ts

```diff
@@ -0,0 +1,34 @@
+export interface APIResponse {
+  success: boolean;
+  data: string[];
+}
+
+export async function getCatImages(): Promise<APIResponse> {
+  const response = await fetch("/api/photos/cats");
+  const { data, success } = await response.json();
+
+  return {
+    success: response.ok && success,
+    data,
+  };
```

Comment:
How would you handle failed requests here?

Reply:
Good question! I'd want the user to know that the request has failed, a generic way to do this is to display some form of Snackbar notification, but here I think a message within the Carousel indicating the error would be more appropriate. In many cases, the API would include a user-readable message with the API response for the failure reason, for instance, a specific message for bad request, so I'd want this message to be displayed as well. 

Design-wise, since handling errors directly from fetch everywhere can get repetitive, I would add a shared fetch utility function that would handle extracting error messages from request response or adding generic user-readable messages based on error codes if a message from the API does not exist, or for outright failure with a try-catch.

And for the Carousel component, I'd add another prop `error: string` which includes an error message in case there's an error, and the error would be displayed as a banner around the bottom of the Carousel. 

I didn't end up doing much error handling here, but happy to implement this!